taxonomy/utils.py

```diff
@@ -234,17 +234,24 @@ def process_skill_attr_text(text_data: str, product_type: ProductTypes) -> dict:
     return extra_data
 
 
-def skip_product_processing(extra_data: dict, key_or_uuid: str, product_type: ProductTypes) -> bool:
+def xblock_with_same_content(extra_data: dict, key_or_uuid: str):
     """
-    Check whether to skip processing.
+    Return identifier of the first xblock with same content that is already tagged.
     """
-    # currently only xblock text is checked
-    if product_type != ProductTypes.XBlock:
-        return False
+    model, identifier = get_product_skill_model_and_identifier(ProductTypes.XBlock)
+    similar_product = model.objects.filter(
+        **{'auto_processed': True, **extra_data}
+    ).exclude(
+        **{identifier: key_or_uuid}
+    ).first()
+    return getattr(similar_product, identifier, None)
 
-    if not extra_data:
-        return True
-    model, identifier = get_product_skill_model_and_identifier(product_type)
+
+def verify_xblock_existence_and_content_changes(extra_data: dict, key_or_uuid: str):
+    """
+    Raise SkipProductProcessingError if product has been already tagged and content has not changed.
+    """
+    model, identifier = get_product_skill_model_and_identifier(ProductTypes.XBlock)
```

Comment:
Will we ever end up in a scenario where an xblock that has already been tagged is returned by the provider again for processing? Aren't the usage keys going to be different even if both xblocks are exactly the same?

Reply:
>Will we ever end up in a scenario where an xblock that has already been tagged is returned by the provider again for processing? 

@sameenfatima78 It is possible that we run the command again for a course which already has some xblocks tagged. Also, whenever a xblock is updated in edx-platform, the smallest unit that modulestore publishes is unit/vertical block. So, it is possible that a new xblock was added in the unit which already has tagged video xblock. The published event will send the unit as the affected block and we will process it along with its children. We don't want ping the external API in case of the tagged children xblocks with no change.

>Aren't the usage keys going to be different even if both xblocks are exactly the same?

Don't think the usage key changes at any point.

Reply:
Got it, this answers my question. Thanks!